src/controllers/hello-world.ts

```diff
@@ -4,16 +4,24 @@
 // License text available at https://opensource.org/licenses/MIT
 'use strict';
 
-import {api, inject} from '@loopback/core';
-import {controllerSpec} from './hello-world.api';
-import {authenticate, AuthenticationBindings, UserProfile} from '@loopback/authentication';
+import { api, inject } from '@loopback/core';
+import { helloWorldApi } from '../apidefs/hello.world.api';
+import { authenticate, AuthenticationBindings, UserProfile } from '@loopback/authentication';
+import { UserRepository } from '../repositories/user.repository';
+import { User } from '../models/user';
 
-@api(controllerSpec)
+@api(helloWorldApi)
 export class HelloWorldController {
+  @inject('repositories.UserRepository') repository: UserRepository
```

Comment:
Property injection should be used for things that are optional and have a reasonable default. Dependencies that a class requires in order to function correctly should be injected via constructor arguments.

The idea is that `new HelloWorldController(...)` should create a fully functional controller instance that does not require any additional setup.